test/integration/DataStorage.spec.ts

```diff
@@ -142,8 +167,44 @@ describe('MessageParticle', () => {
             .subscribe({
                 complete: () => {
                     // Look for the data in both accounts
-                    const acc1stream = identity1.account.dataSystem.getApplicationData(appId).pipe(filter(update => update.data.payload.data === payload))
-                    const acc2stream = identity2.account.dataSystem.getApplicationData(appId).pipe(filter(update => update.data.payload.data === payload))
+                    const acc1stream = identity1.account.dataSystem.getApplicationData(appId).pipe(
+                        filter(update => update.data.payload.data === payload),
+                    )
+                    const acc2stream = identity2.account.dataSystem.getApplicationData(appId).pipe(
+                        filter(update => update.data.payload.data === payload),
+                    )
+
+                    // Finish only when data has been found in both accounts
+                    zip(acc1stream, acc2stream).subscribe(val => {
+                        done()
+                    })
```

Comment:
This test would time-out (and thereby fail) if `acc2stream` wouldn't see the data sent by `acc1stream` right? What is the default time-out threshold set to?

Would it be an option to, instead of filtering out non-matching payload data, map it to boolean `true` or `false` and then call `done()` depending on that boolean flag? Something along the lines:

```javascript
const acc1stream = identity1.account.dataSystem
        .getApplicationData(appId)
        .pipe(map(update => update.data.payload.data === payload))

const acc2stream = identity2.account.dataSystem
        .getApplicationData(appId)
        .pipe(map(update => update.data.payload.data === payload))

zip(acc1stream, acc2stream).subscribe(val => {
        if (val[0] === val[1] === true) {
            done()
        }
})
```

This way wouldn't have to wait for a timeout to fail our tests.

Reply:
You can specify the timeout using `this.timeout(50000)` in the test case, the default is 2s I think.

It could be written that way, but I don't like that it makes an assumption that there will never be any other values that show up for some reason, since that shouldn't affect the success of this test necessarily, but it would probably be fine in this case.

And it would also timeout anyway if either of the accounts don't get any values, which is the most likely failure case.

Reply:
Good point!